test/e2e/networking.go

```diff
@@ -101,155 +102,212 @@ var _ = Describe("Networking", func() {
-				},
-			},
-			Spec: api.ServiceSpec{
-				Ports: []api.ServicePort{{
-					Protocol:   "TCP",
-					Port:       8080,
-					TargetPort: util.NewIntOrStringFromInt(8080),
-				}},
-				Selector: map[string]string{
-					"name": svcname,
-				},
-			},
-		})
-		if err != nil {
-			Failf("unable to create test service named [%s] %v", svc.Name, err)
-		}
+	// TODO: support DNS on vagrant #3580
+	for _, svcSoak := range serviceSoakTests {
+		//copy to local to avoid range overwriting
+		timeout := svcSoak.timeout
+		serviceNum := svcSoak.service
+		It(fmt.Sprintf("should function for intrapod communication between all hosts in %v parallel services", serviceNum),
+			func(doneTimeout Done) {
+				SkipIfProviderIs("vagrant")
+				Logf("running service test with timeout = %v for %v", timeout, serviceNum)
+				runNetTest(doneTimeout, f, makePorts(serviceNum), nettestVersion)
+			}, timeout)
+	}
 
-		// Clean up service
-		defer func() {
-			defer GinkgoRecover()
-			By("Cleaning up the service")
-			if err = f.Client.Services(f.Namespace.Name).Delete(svc.Name); err != nil {
-				Failf("unable to delete svc %v: %v", svc.Name, err)
-			}
-		}()
+})
 
-		By("Creating a webserver (pending) pod on each node")
+//PeerStatus will either fail, pass, or continue polling
+func pollPeerStatus(ch chan int, f *Framework, svc *api.Service) bool {
+
+	Logf("Begin polling " + svc.Name)
+	getDetails := func() ([]byte, error) {
+		return f.Client.Get().
+			Namespace(f.Namespace.Name).
+			Prefix("proxy").
+			Resource("services").
+			Name(svc.Name).
+			Suffix("read").
+			DoRaw()
+	}
 
-		nodes, err := f.Client.Nodes().List(labels.Everything(), fields.Everything())
+	getStatus := func() ([]byte, error) {
+		return f.Client.Get().
+			Namespace(f.Namespace.Name).
+			Prefix("proxy").
+			Resource("services").
+			Name(svc.Name).
+			Suffix("status").
+			DoRaw()
+	}
+
+	passed := false
+
+	expectNoError(wait.Poll(2*time.Second, pollTimeout*time.Second, func() (bool, error) {
+		body, err := getStatus()
 		if err != nil {
 			Failf("Failed to list nodes: %v", err)
 		}
-		// previous tests may have cause failures of some nodes. Let's skip
-		// 'Not Ready' nodes, just in case (there is no need to fail the test).
-		filterNodes(nodes, func(node api.Node) bool {
-			return isNodeReadySetAsExpected(&node, true)
-		})
 
-		if len(nodes.Items) == 0 {
-			Failf("No Ready nodes found.")
-		}
-		if len(nodes.Items) == 1 {
-			// in general, the test requires two nodes. But for local development, often a one node cluster
-			// is created, for simplicity and speed. (see issue #10012). We permit one-node test
-			// only in some cases
-			if !providerIs("local") {
-				Failf(fmt.Sprintf("The test requires two Ready nodes on %s, but found just one.", testContext.Provider))
+		// Finally, we pass/fail the test based on if the container's response body, as to whether or not it was able to find peers.
+		switch {
+		case string(body) == "pass":
+			Logf("Passed.")
```

Comment:
the log messages in this loop seem very spammy and unhelpful as the number of services increases